packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2702,9 +2702,11 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     final Offset finalPosition = renderEditable.getLocalRectForCaret(_lastTextPosition!).centerLeft - _floatingCursorOffset;
     if (_floatingCursorResetController!.isCompleted) {
       renderEditable.setFloatingCursor(FloatingCursorDragState.End, finalPosition, _lastTextPosition!);
-      if (_lastTextPosition!.offset != renderEditable.selection!.baseOffset) {
+      // Only change if new position is out of current selection range, as the
+      // selection may have been modified using the iOS keyboard selection gesture.
+      if (_lastTextPosition!.offset < renderEditable.selection!.start || _lastTextPosition!.offset >= renderEditable.selection!.end) {
```

Comment:
Looks like this changes the behavior of the floating cursor when the initial selection is not empty if the floating cursor lands within the selected range?

Reply:
In a UIKit app the floating cursor removes the selection on start it seems

Reply:
The selection is being cleared in flutter apps too by UIKit when floating cursor starts, so there shouldn't be any change.

Reply:
Ah it already does that without the engine PR?

Reply:
Yeah, I just checked in a flutter app built without the PR